tests/lib/fakestore/store/store.go

```diff
@@ -134,12 +129,18 @@ func (s *Store) Start() error {
 // Stop stops the server
 func (s *Store) Stop() error {
 	timeoutTime := 2000 * time.Millisecond
-	s.srv.Stop(timeoutTime / 2)
-
-	select {
-	case <-s.srv.StopChan():
-	case <-time.After(timeoutTime):
-		return fmt.Errorf("store failed to stop after %s", timeoutTime)
+	closedC := make(chan error, 1)
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutTime)
+	defer cancel()
+	go func() {
+		closedC <- s.srv.Shutdown(ctx)
+	}()
+
+	err := <-closedC
+	if err != nil {
+		// forceful close
+		s.srv.Close()
+		return fmt.Errorf("store failed to stop after: %s", timeoutTime)
 	}
```

Comment:
Perhaps I'm missing something, but why is `closedC` necessary at all? Seems like we create the channel just to wait for `s.src.Shutdown(ctx)`. Why can't we call `s.src.Shutdown(ctx)` synchronously instead of `err := <-closedC`?

Reply:
Good point. I started with select like it was in the original code, but then added a context and forgot to drop to goroutine completely.